seedtool/userinterface.ino

```diff
@@ -1397,6 +1491,277 @@ void enter_share() {
+
+          const char * title = "Enter derivation path:";
+          int yy = 18;
+
+          g_display->setFont(&FreeSansBold9pt7b);
+          Point p = text_center(title);
+          g_display->setCursor(p.x, yy);
+          g_display->println(title);
+
+          g_display->setFont(&FreeMonoBold12pt7b);
+          g_display->setCursor(x_off, yy + 50);
+          g_display->println(path_start + path_entered);
+
+          if (path_is_valid)
+              is_valid_tip = "Valid";
+          else
+              is_valid_tip = "Invalid";
+          p = text_center(is_valid_tip.c_str());
+          g_display->setCursor(p.x, p.y + 20);
+          g_display->println(is_valid_tip);
+
+          yy = 195; // Absolute, stuck to bottom
+          g_display->setFont(&FreeMono9pt7b);
+          String right_option = "Ok #";
+          int x_r = text_right(right_option.c_str());
+          g_display->setCursor(x_r, yy);
+          g_display->println(right_option);
+
+          String left_option = "Cancel *";
+          g_display->setCursor(0, yy);
+          g_display->println(left_option);
+
+          yy -= 25;
+          String tip = "A=/  B=h";
+          g_display->setCursor(0, yy);
+          g_display->println(tip);
+
+          right_option = "Del D";
+          x_r = text_right(right_option.c_str());
+          g_display->setCursor(x_r, yy);
+          g_display->println(right_option);
+      }
+      while (g_display->nextPage());
+
+      char key;
+      do {
+          key = g_keypad.getKey();
+      } while (key == NO_KEY);
+
+      switch (key) {
+        case '#':
+            if (path_is_valid) {
+                keystore.set_derivation_path(path_start + path_entered);
+                g_uistate = XPUB_MENU;
+                return;
+            }
+            break;
+        case '0': case '1': case '2': case '3': case '4':
+        case '5': case '6': case '7': case '8': case '9':
+            path_entered += key;
+            break;
+        case 'A':
+            path_entered += "h";
+            break;
+        case 'B':
+            path_entered += "/";
+            break;
+        case 'D':
+            if (path_entered.length() > 0)
+                path_entered.remove(path_entered.length()-1);
+            break;
+        case '*':
+            g_uistate = XPUB_MENU;
+            return;
+        default:
+            break;
+      }
+
+      if (keystore.get_xpub((path_start + path_entered).c_str(), &keystore.root, &xpubkey)) {
+          path_is_valid = true;
+      }
+      else {
+          path_is_valid = false;
+      }
+    }
+}
+
+void xpub_menu(void) {
+    int x_off = 5;
+    bool option_slip132 = false;
+    bool option_derivation_path = false;
+
+    while (true) {
+      g_display->firstPage();
+      do
+      {
+          g_display->setPartialWindow(0, 0, 200, 200);
+          g_display->fillScreen(GxEPD_WHITE);
+          g_display->setTextColor(GxEPD_BLACK);
+
+          const char * title = "Xpub Menu";
```

Comment:
This menu is a little confusing to me, initially.  It would be great if it clearly said:
"Here is what is currently selected" and
"Here are changes you can make"

Maybe it could show the current deriv path and the current state of the two options and then below:
A: change deriv path
B: add (or remove) SLIP32 version
C: add (or remove) Deriv Path

I'm open to others opinions here, UX is not my forte.  @ChristopherA Thoughts?

Reply:
Fixed